### PR TITLE
US62152 - add pep8 lint 

### PR DESCRIPTION
**Pull request by daniel-greening**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,2 +1,6 @@
 [metadata]
 description-file = share/README.md
+
+[pycodestyle]
+ignore = W601
+max-line-length = 160
```
Comment (anton-b) [inline setup.cfg]: max line length we agreed on is 120. There should be code style guide somewhere(it is for erlang, but i guess this point is valid for any of our code). Ofcourse it is discussible.

File: alertlogic/dynapi.py [hunk 2]
```diff
@@ -151,34 +168,43 @@ def call(self, session, url_args, json=None):
...
+
 
 def substitute_path_args(path, args):
     parts = [part for part in path.lower().split("/") if len(part) > 0]
     substituted = ""
     for part in parts:
         if part.startswith(":"):
-            required_arg = part[1:] # removes ":" at the beginning of the string
+            required_arg = part[
+                1:]  # removes ":" at the beginning of the string
```
Comment (anton-b) [inline alertlogic/dynapi.py]: what is the reason for such weird line break, it seems that pycodestyle is not complaining if it is on a single line.

File: alertlogic/dynapi.py [hunk 3]
```diff
@@ -151,34 +168,43 @@ def call(self, session, url_args, json=None):
...
 
+
 def substitute_query_args(query, args):
     parsed = urlparse.parse_qs(query)
     for key in parsed:
         value = parsed[key][0]
         if value.startswith(":"):
-            required_arg = value[1:] # removes ":" at the beginning of the string
+            required_arg = value[
+                1:]  # removes ":" at the beginning of the string
```
Comment (anton-b) [inline alertlogic/dynapi.py]: same here

File: alertlogic/dynapi.py [hunk 4]
```diff
@@ -151,34 +168,43 @@ def call(self, session, url_args, json=None):
...
             if required_arg in args:
                 parsed[key] = args[required_arg]
             else:
-                raise InvalidEndpointCall("missing required url argument {}".format(required_arg))
+                raise InvalidEndpointCall(
+                    "missing required url argument {}".format(required_arg))
```
Comment (anton-b) [inline alertlogic/dynapi.py]: same

File: alertlogic-cli [hunk 5]
```diff
@@ -9,12 +9,15 @@ import traceback
```
Comment (anton-b) [inline alertlogic-cli]: ran `make lint` it didn't notice changes in alertlogic-cli. When i do `pycodestyle alertlogic-cli`:
```
pycodestyle alertlogic-cli
alertlogic-cli:7:15: E401 multiple imports on one line
alertlogic-cli:25:31: E401 multiple imports on one line
alertlogic-cli:33:36: E401 multiple imports on one line
```

File: alertlogic-cli [hunk 6]
```diff
@@ -9,12 +9,15 @@ import traceback
...
 
+
 def import_dependencies():
     try:
         global requests
         import requests
     except ImportError:
-        print("requests library required, install it first: pip install requests")
+        print(
+            "requests library required, install it first: pip install requests"
```
Comment (anton-b) [inline alertlogic-cli]: It is kind of questionable formatting. But besides that, it is inconsistent with L27 and L36.

File: alertlogic-cli [hunk 7]
```diff
@@ -27,70 +30,96 @@ def import_dependencies():
...
+        "--deployment_id",
+        help="deployment id used by most commands (uuid)")
+
+    subparsers = parser.add_subparsers(
+        title="commands", description="valid commands")
+
     for command in commands:
-        subparser = subparsers.add_parser(command["name"], help=command["help"])
+        subparser = subparsers.add_parser(
+            command["name"], help=command["help"])
```
Comment (anton-b) [inline alertlogic-cli]: imo, breaking line this way when it is not really required is not adding to readability(i might be wrong in python world, so we can discuss it).

File: alertlogic-cli [hunk 8]
```diff
@@ -27,70 +30,96 @@ def import_dependencies():
...
-    
+
     if args.logging_config_file:
         logging.config.fileConfig(args.logging_config_file)
-    
+
     config = alertlogiccli.config.Config(args.config_file, args.profile)
-    credentials = alertlogiccli.credentials.Credentials(args.credentials_file, args.profile)
-    
+    credentials = alertlogiccli.credentials.Credentials(
```
Comment (anton-b) [inline alertlogic-cli]: same here as L98

File: alertlogic/auth.py [hunk 9]
```diff
@@ -33,18 +37,22 @@ def _authenticate(self, username, password):
...
 
         try:
             self._token = response.json()["authentication"]["token"]
         except (KeyError, TypeError, ValueError):
             raise AuthenticationException("token not found in response")
 
         try:
-            self.account_id = response.json()["authentication"]["account"]["id"]
+            self.account_id = response.json()["authentication"]["account"][
+                "id"]
```
Comment (anton-b) [inline alertlogic/auth.py]: weird line break

File: alertlogic/auth.py [hunk 10]
```diff
@@ -33,18 +37,22 @@ def _authenticate(self, username, password):
...
             auth = requests.auth.HTTPBasicAuth(username, password)
-            response = requests.post(self.region.get_api_endpoint()+"/aims/v1/authenticate", auth=auth)
+            response = requests.post(
+                self.region.get_api_endpoint() + "/aims/v1/authenticate",
+                auth=auth)
             response.raise_for_status()
         except requests.exceptions.HTTPError as e:
-            raise AuthenticationException("invalid http response {}".format(e.message))
+            raise AuthenticationException("invalid http response {}".format(
+                e.message))
```
Comment (anton-b) [inline alertlogic/auth.py]: weird line break

File: alertlogic/dynapi.py [hunk 11]
```diff
@@ -99,15 +110,19 @@ def __getattr__(self, name):
         this is what allows us to dynamically add functions to an existing object
         """
         if name in self._endpoints:
+
             def handler(*args, **kwargs):
                 # json argument is passed separately
                 json = kwargs.pop("json", None)
-                return self._endpoints[name].call(self._session, kwargs, json=json)
+                return self._endpoints[name].call(
+                    self._session, kwargs, json=json)
```
Comment (anton-b) [inline alertlogic/dynapi.py]: weird line break

File: alertlogic/dynapi.py [hunk 12]
```diff
@@ -151,34 +168,43 @@ def call(self, session, url_args, json=None):
         :param url_args: dict with values to replace url parameters, see parse_args() for more info
         :param json: this will be passed adhoc to requests
         """
-        parsed_url = session.region.get_api_endpoint() + self.parse_url(url_args)
-        log.debug("calling requests: operation={} url={} json={}".format(self.operation, parsed_url, json))
-        return requests.request(self.operation, parsed_url, json=json, auth=session)
+        parsed_url = session.region.get_api_endpoint() + self.parse_url(
+            url_args)
```
Comment (anton-b) [inline alertlogic/dynapi.py]: weird line break

**Comment by anton-b** [comment]
@daniel-greening the only major thing is that check is not runing over alertlogic-cli. But i also would like to discuss why line breaks are that weird sometimes. As i said i might not know about some python cs standards, will be happy if you share it with me tomorrow 👍

File: setup.cfg (hunk 1 above)
Comment (daniel-greening) [reply setup.cfg]: Updated to 120 and added config to allow linting of alertlogic-cli

File: alertlogic-cli (hunk 5 above)
Comment (daniel-greening) [reply alertlogic-cli]: Fixed in setup.cfg

File: alertlogiccli/credentials.py [hunk 13]
```diff
@@ -17,21 +20,24 @@ class Credentials():
...
+        del (self._password)
         return session
 
     def _read(self, filename):
         try:
             read_ok = self._parser.read(filename)
-            if not filename in read_ok:
-                raise CredentialsException("unable to read {}".format(filename))
+            if filename not in read_ok:
+                raise CredentialsException(
```
Comment (anton-b) [inline alertlogiccli/credentials.py]: still weird

File: alertlogiccli/credentials.py [hunk 14]
```diff
@@ -40,6 +46,9 @@ def _set_profile(self, profile):
             self._username = self._parser.get(profile, "username")
             self._password = self._parser.get(profile, "password")
         except ConfigParser.NoSectionError as e:
-            raise CredentialsException("credentials profile {} not found".format(profile))
+            raise CredentialsException(
+                "credentials profile {} not found".format(profile))
```
Comment (anton-b) [inline alertlogiccli/credentials.py]: it is definetely less than 120, should not be broken.

File: alertlogiccli/credentials.py [hunk 15]
```diff
@@ -40,6 +46,9 @@ def _set_profile(self, profile):
...
         except ConfigParser.NoOptionError as e:
-            raise CredentialsException("credentials field {} not found in profile {}".format(e.option, profile))
+            raise CredentialsException(
+                "credentials field {} not found in profile {}".format(
```
Comment (anton-b) [inline alertlogiccli/credentials.py]: weird break and weird identation

File: alertlogic/dynapi.py (hunk 2 above)
Comment (anton-b) [reply alertlogic/dynapi.py]: not fixed yet

File: alertlogiccli/commands/deployment/resources.py [hunk 16]
```diff
@@ -5,22 +5,28 @@
...
         parser.set_defaults(command=self)
 
     def execute(self, context):
         args = context.get_final_args()
         launcher = context.get_services().launcher
+
         try:
-            response = launcher.getawsresourcesbyenvironment(account_id=args["account_id"],
-                                                             environment_id=args["deployment_id"])
+            response = launcher.getawsresourcesbyenvironment(
```
Comment (anton-b) [inline alertlogiccli/commands/deployment/resources.py]: inconsistent with others - bottom bracket to be on the separate line

File: alertlogiccli/commands/deployment/scan_queue.py [hunk 17]
```diff
@@ -3,27 +3,33 @@
 import requests
 import json
 
+
 class ListScanQueues(alertlogiccli.command.Command):
     """List hosts in scan queues for a given deployment"""
+
     def configure_parser(self, subparsers):
-        parser = subparsers.add_parser("list_scan_queues", help="lists hosts in scan queues for a given deployment")
+        parser = subparsers.add_parser(
```
Comment (anton-b) [inline alertlogiccli/commands/deployment/scan_queue.py]: bottom bracket on the sep. line

File: alertlogiccli/commands/deployment/scan_queue.py [hunk 18]
```diff
@@ -3,27 +3,33 @@
...
+            help="lists hosts in scan queues for a given deployment")
         parser.set_defaults(command=self)
 
     def execute(self, context):
         args = context.get_final_args()
         scan_scheduler = context.get_services().scan_scheduler
         try:
-            response = scan_scheduler.listscanassets(account_id=args["account_id"],
-                                                     environment_id=args["deployment_id"])
+            response = scan_scheduler.listscanassets(
```
Comment (anton-b) [inline alertlogiccli/commands/deployment/scan_queue.py]: bottom bracket

File: alertlogiccli/commands/deployment/scan_queue.py [hunk 19]
```diff
@@ -36,19 +42,25 @@ def group_by_vpc(self, assets):
...
 class ScanHost(alertlogiccli.command.Command):
     """Puts a host to the immediate scan queue"""
+
     def configure_parser(self, subparsers):
         parser = subparsers.add_parser("scan_host", help="puts a host in the immediate scan queue")
         parser.set_defaults(command=self)
-        
-        parser.add_argument("--host_key", dest="asset_key", required=True, help="a host key to put in the queue")
+
+        parser.add_argument(
```
Comment (anton-b) [inline alertlogiccli/commands/deployment/scan_queue.py]: bottom bracket

File: alertlogiccli/commands/deployment/scan_queue.py [hunk 20]
```diff
@@ -36,19 +42,25 @@ def group_by_vpc(self, assets):
...
         args = context.get_final_args()
         scan_scheduler = context.get_services().scan_scheduler
         try:
-            response = scan_scheduler.scanasset(account_id=args["account_id"],
-                                                environment_id=args["deployment_id"],
-                                                asset_key=args["asset_key"])
+            response = scan_scheduler.scanasset(
+                account_id=args["account_id"],
+                environment_id=args["deployment_id"],
+                asset_key=args["asset_key"])
```
Comment (anton-b) [inline alertlogiccli/commands/deployment/scan_queue.py]: bottom backet, please look through the other files for the same problem.

File: alertlogiccli/credentials.py [hunk 21]
```diff
@@ -5,9 +5,12 @@
 
 log = logging.getLogger()
 
+
 class CredentialsException(Exception):
     def __init__(self, message):
-        super(CredentialsException, self).__init__("credentials error: {}".format(message))
+        super(CredentialsException,
```
Comment (anton-b) [inline alertlogiccli/credentials.py]: breaking line which is less than 120?

File: tests/auth_tests.py [hunk 22]
```diff
@@ -7,36 +7,56 @@
...
+                "id": "ACCOUNT_ID"
+            }
+        }
+    }
+    httpretty.register_uri(
+        httpretty.POST,
+        "http://mock/aims/v1/authenticate",
+        status=200,
+        content_type="text/json",
+        body=json.dumps(response_body))
```
Comment (anton-b) [inline tests/auth_tests.py]: bottom bracket

File: tests/dynapi_tests.py [hunk 23]
```diff
@@ -9,30 +9,37 @@
...
 
+
 def test_parse_apidoc():
     with mock.patch.object(alertlogic.dynapi, 'API_SERVICES', "") as mock_value:
         services = alertlogic.dynapi.Services()
-        service_data = [
-            {"name": "test_endpoint1", "type": "get", "url": "/test_service/endpoint1" },
-            {"name": "test_endpoint2", "type": "post", "url": "/test_service/endpoint2" }
-        ]
+        service_data = [{
```
Comment (anton-b) [inline tests/dynapi_tests.py]: i'd make it as follows:
```
[
    {
      a,
      b
    },
   {
      c, 
      d
   }
]
```

**Pull request merged**